trie/stackproof.go

```diff
@@ -459,7 +476,7 @@ func (g *GenerativeTrie) UpdateAll(keys []common.Hash, values [][]byte) error {
 		g.stack.Update(keys[i][:], values[i])
 	}
 	last := keys[len(keys)-1]
-	g.rightHandBorder = last[:]
+	g.rightHandBorder = keybytesToHex(last[:])
```

Comment:
This migth be wrong, in fact, it probably is. Look into this, make a test for it